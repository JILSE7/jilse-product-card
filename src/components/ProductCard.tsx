import styles from '../styles/styles.module.css';


import React, { createContext } from 'react';
import useCount from '../hooks/useCount';
import { IProductContext, IPropsProductCardComponent } from '../interfaces/ProductInterfaces';



export const ProductContext = createContext({} as IProductContext);
const {Provider} = ProductContext;

export const ProductCard = ({product, children, className, style, onChange, value, initialValues}:IPropsProductCardComponent) => {
    
  

  const {counter, increasyBy, maxCount, isMaxCountReached, reset} = useCount({onChange, product,value, initialValues});

  return (
    <Provider value={{counter, increasyBy, product, maxCount}}>
      
        <div className={`${styles.productCard} ${className}`} style={style}>

            {
           
                  children({count:counter,increasyBy,isMaxCountReached,maxCount: maxCount!, reset, product})
               
            }

        </div>

    </Provider>
  )
}

//*OTRA FORMA DE EXPORTAR LOS COMPONENTES PARA QUE SE MIREN DE LA SIGUIENTE FORMA
//?PASAMOS DE ESTO, QUE NO ESTA MAL, SOLO ES UNA FORMA 
{/* <ProductCard product={product}>
  <ProductImage/>
  <ProductTitle title="Coffe Cup"/>
  <ProducButtons counter={counter} increasyBy={increasyBy}/>
</ProductCard> */}

//?PASAMOS A PONERLE MAS PROPIEDADES AL COMPONENTE
 //* <ProductCard product={product}>
      //?<ProductCard.Image/>
      //?<ProductCard.Title title="Coffe Cup"/>
      //?<ProductCard.Buttons counter={counter} increasyBy={increasyBy}/>
//* </ProductCard> 

//*ASIGNAMOS LAS PROPIEDADES


