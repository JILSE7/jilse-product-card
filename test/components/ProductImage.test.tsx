

import React from 'react';
//import * as ReactDOM from 'react-dom';
import rendered from 'react-test-renderer';
import  ProductCard, {ProductImage}  from '../../src/components';

import { product2 } from '../data/products';




describe('Pruebas en el product image', () => {

    test('should show component with personalize title', () => { 
        
        const wrapper = rendered.create(
            <ProductImage img='https://hola.jpg' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();

    })

    test('debe de mostrar el componente la imagen del producto', () => {
          
        const wrapper = rendered.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <>
                              <ProductImage className='said'/>
                        </>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})