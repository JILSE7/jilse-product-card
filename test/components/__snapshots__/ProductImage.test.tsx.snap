// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el product image debe de mostrar el componente la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Coffe mug"
    className="productImg said"
    src="/coffee-mug.png"
  />
</div>
`;

exports[`Pruebas en el product image should show component with personalize title 1`] = `
<img
  alt="Coffe mug"
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
